docs/source/api.rst

```diff
@@ -125,3 +125,67 @@ For example::
 
 
 
+Detailed API reference
+------------------------
+Module: :mod:`traitlets.traitlets`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. automodule:: traitlets.traitlets
+    :members:
+    :private-members:
```

Comment:
I have to disagree with that.
Marking a method as private, conveys a strong enough signal to the consumer of the API, but we should not limit in addition her capability to understand the library. 

Sometimes reading the documentation (or looking at the sources) is needed to understand the operation of a higher-level function. For instance, I had to read `Configurable._load_config_files(filename, path)` to see that the `path` arg in `Configurable.load_config_file(filename.path)` can be a list.